test/index.test.ts

```diff
@@ -293,7 +390,116 @@ describe('HipThrusTS', () => {
     // @todo: add test coverage for TRIPLE-OPERATOR pipes
     // @todo: add test coverage for QUADRUPLE-AND-MORE-OPERATOR pipes (use empty objects)
     // @todo: add test coverage for multi-stage operators
```

Comment:
These are supposed to be the unit tests I was talking about right?
If so...
- [ ] simplify.  you're testing HTPipeTest itself - we don't need/want any left/right or even HTPipe.  Just send it an already built reference object. 
- [ ] You're testing types, but you should also test function.  Give HTPipe data with the CORRECT types but INCORRECT data to make sure it FAILS in that case.

For the type tests, I can't follow the strategy.  The things to think about are:
1. pipeIn/pipeOut
2. test for extra key exists on pipeIn (or pipeOut) compared to pipe
3. test for key missing on pipeIn (or pipeOut) compared to pipe
4. test for keys the same but type mismatch
So all-in-all I expect 2x3=6 tests for the types
- [ ] all 6 type failure cases

Reply:
done